astronomy/orbital_elements_test.cpp

```diff
@@ -0,0 +1,422 @@
+    astronomy::OrbitalElements::EquinoctialElements const& elements) {
+  return ToMathematica(std::make_tuple((elements.t - J2000) / Second,
+                                       elements.a / Metre,
+                                       elements.h,
+                                       elements.k,
+                                       elements.λ / Radian,
+                                       elements.p,
+                                       elements.q,
+                                       elements.pʹ,
+                                       elements.qʹ));
+}
+}  // namespace internal_mathematica
+}  // namespace mathematica
+
+namespace astronomy {
+
+using base::make_not_null_unique;
+using base::not_null;
+using base::OFStream;
+using geometry::Instant;
+using geometry::Position;
+using geometry::Velocity;
+using integrators::EmbeddedExplicitRungeKuttaNyströmIntegrator;
+using integrators::SymmetricLinearMultistepIntegrator;
+using integrators::methods::DormandالمكاوىPrince1986RKN434FM;
+using integrators::methods::QuinlanTremaine1990Order12;
+using physics::BodyCentredNonRotatingDynamicFrame;
+using physics::DegreesOfFreedom;
+using physics::DiscreteTrajectory;
+using physics::Ephemeris;
+using physics::KeplerianElements;
+using physics::KeplerOrbit;
+using physics::MassiveBody;
+using physics::MasslessBody;
+using physics::OblateBody;
+using physics::RotatingBody;
+using physics::SolarSystem;
+using quantities::Angle;
+using quantities::AngularFrequency;
+using quantities::Cos;
+using quantities::Pow;
+using quantities::Sqrt;
+using quantities::Time;
+using quantities::astronomy::JulianYear;
+using quantities::si::ArcMinute;
+using quantities::si::ArcSecond;
+using quantities::si::Day;
+using quantities::si::Degree;
+using quantities::si::Kilo;
+using quantities::si::Metre;
+using quantities::si::Micro;
+using quantities::si::Milli;
+using quantities::si::Minute;
+using quantities::si::Radian;
+using quantities::si::Second;
+using testing_utilities::AbsoluteError;
+using testing_utilities::AlmostEquals;
+using testing_utilities::IsNear;
+using testing_utilities::IsOk;
+
+class OrbitalElementsTest : public ::testing::Test {
+ protected:
+  OrbitalElementsTest()
+      : solar_system_(
+            SOLUTION_DIR / "astronomy" / "sol_gravity_model.proto.txt",
+            SOLUTION_DIR / "astronomy" /
+                "sol_initial_state_jd_2451545_000000000.proto.txt"),
+        real_ephemeris_(solar_system_.MakeEphemeris(
+            /*accuracy_parameters=*/{/*fitting_tolerance=*/1 * Milli(Metre),
+                                     /*geopotential_tolerance=*/0x1p-24},
+            Ephemeris<ICRS>::FixedStepParameters(
+                SymmetricLinearMultistepIntegrator<QuinlanTremaine1990Order12,
+                                                   Position<ICRS>>(),
+                /*step=*/10 * Minute))),
+        real_earth_(*solar_system_.massive_body(*real_ephemeris_, "Earth")),
+        oblate_earth_ephemeris_([this] {
```

Comment:
Please don't use unreadable lambdas like this.  I have no idea what this is doing.  Do the initialization in the body of the constructor.

Reply:
OBE.